src/JBGovernanceNFT.sol

```diff
@@ -59,59 +56,55 @@ contract JBGovernanceNFT is ERC721Votes, ReentrancyGuard {
     /**
      * @dev Mint a new NFT and stake tokens.
      * @param _mints An array of struct containing the beneficiary and stake amount for each NFT to be minted.
-     * @return _tokenId The token ID of the newly minted NFT.
+     * @return _tokenId The token IDs of the newly minted NFTs.
     */
-    function mint_and_stake(JBGovernanceNFTMint[] calldata _mints) external nonReentrant returns (uint256 _tokenId) {
+    function mint(JBGovernanceNFTMint[] calldata _mints) external returns (uint256[] memory) {
```

Comment:
looks good just a couple of things in mint
1. I feel we should have a reentrancy check wdyt of it ?
2. also feel e can follow checks and effects in mint and do the transfer at the end after every storage update wdyt of that ?

Reply:
1: We can have a reentrancy check, but if we can prevent reentrency from being able to do any damage thats even better, as it saves some gas
2: I think checks and effects is a good pattern, but I feel like the `safeTransferFrom` is more of a check than an effect, by doing it before the mint and storage updates we make it so if a contract wants to reenter they need to have the funds to mint multiple nfts. So we are using it as a check for the users balance. 

wdyt, does that make sense?

Reply:
the only thing I mentioned to have a reentrancy check was because safeMint also has a onErc721Received hook 
besides that I'm okay if we don't have it for now 
we can think about it during the audit stage

Reply:
I replaced `_safeMint` with regular `_mint`, that way there is no reentrency risk and it saves some more gas, wdyt

Reply:
yeah cool
lfg